lib/addressfinder/http.rb

```diff
@@ -15,11 +15,16 @@ def start(&block)
       end
     end
 
-    def request(args)
+    def request(request_uri)
+      raise ArgumentError, "request_uri is nil" if request_uri.nil?
```

Comment:
Yeah, you're right, looking into it; URI has a `raise Argument` error it throws. I added the `request_uri is nil` as an early exit strategy bu since we are not checking all edge cases as you said we might as well let the URI handle the `nils` as well.

```ruby
  def URI(uri)
    if uri.is_a?(URI::Generic)
      uri
    elsif uri = String.try_convert(uri)
      URI.parse(uri)
    else
      raise ArgumentError,
        "bad argument (expected URI object or URI string)"
    end
  end
```